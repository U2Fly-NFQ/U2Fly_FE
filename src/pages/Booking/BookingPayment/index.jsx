import { Form, Radio, Checkbox } from 'antd'
import React, { useEffect, useState } from 'react'
import { useLoadingContext } from 'react-router-loading'
import { ButtonOfPage } from '../../../components'
import './index.scss'
export default function BookingPayment() {
  const [value, setValue] = useState(1)
  const [dataBooking, setDataBooking] = useState()
  const loadingContext = useLoadingContext()
  useEffect(() => {
    setDataBooking(dataPayment[value])
    loading()
  }, [value])
  const onChange = (e) => {
    setValue(e.target.value)
  }

  const loading = async () => {
    loadingContext.done()
  }

  let dataPayment = [
    {
      paymentMethod: 'Paypal',
      render: (
        <>
          <Form.Item
            name="agreement"
            valuePropName="checked"
            style={{ marginTop: '10px' }}
            rules={[
              {
                validator: (_, value) =>
                  value
                    ? Promise.resolve()
                    : Promise.reject(new Error('Should accept agreement')),
              },
            ]}
          >
            <Checkbox>
              I have read the <a href="">agreement</a>
            </Checkbox>
          </Form.Item>
          <Form.Item style={{ marginTop: '20px' }}>
            <ButtonOfPage title={'Pay Now'} />
          </Form.Item>
        </>
      ),
    },
  ]

  return (
    <>
      <div className="booking-page__container__item__title">
        <h2>Payemnt methods</h2>
      </div>
      <div className="payment-booking">
        <Radio.Group onChange={onChange} value={value}>
          {dataPayment &&
            dataPayment.map((item, index) => {
              return (
                <Radio value={index} key={index}>
                  {item.paymentMethod}
                </Radio>
              )
            })}
        </Radio.Group>
        <div className="BookingPayment-content">
          {dataBooking && dataBooking.render}
        </div>
      </div>
    </>
  )
}
