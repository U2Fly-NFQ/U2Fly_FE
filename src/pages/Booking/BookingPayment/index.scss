.BookingPayment-content {
  padding-top: 20px;
}
.payment-booking {
  width: 100%;
  &__radio {
    width: 100%;
  }
}
