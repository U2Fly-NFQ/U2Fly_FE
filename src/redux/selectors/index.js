export * from './auth'
export * from './bookingFlightSelector'
export * from './ticket'
