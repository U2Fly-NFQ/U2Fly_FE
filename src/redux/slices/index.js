export * from './searchSlice'
export * from './filterSlice'
export * from './flightSlice'
