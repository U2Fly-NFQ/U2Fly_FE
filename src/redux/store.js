import { configureStore } from '@reduxjs/toolkit'
import bookingFlightsSlice from './slices/bookingFlightsSlice'

// import loginSlice from './slices/loginSlice'
import flightSlice from './slices/flightSlice'
import airportSlice from './slices/airportSlice'

const store = configureStore({
  reducer: {
    // login: loginSlice.reducer,
    bookingFlight: bookingFlightsSlice.reducer,
    flights: flightSlice.reducer,
    airports: airportSlice.reducer,
  },
})

export default store
