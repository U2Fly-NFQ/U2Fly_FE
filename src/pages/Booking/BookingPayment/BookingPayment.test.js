import { I18nextProvider } from 'react-i18next'
import i18n from '../../../translations'
import React, { Suspense } from 'react'
import { store } from '../../../redux/store'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import BookingPayment from './'

describe('Booking Passenger bar test', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <I18nextProvider i18n={i18n}>
            <Router>
              <Suspense>
                <BookingPayment />
              </Suspense>
            </Router>
          </I18nextProvider>
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should finish', async () => {
    render(
      <Provider store={store}>
        <Router>
          <Suspense>
            <BookingPayment />
          </Suspense>
        </Router>
      </Provider>
    )

    const radioButton = screen.getByDisplayValue('0')
    fireEvent.click(radioButton)
  })
})
