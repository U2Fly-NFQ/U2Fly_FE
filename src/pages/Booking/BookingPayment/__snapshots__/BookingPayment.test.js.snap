// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking Passenger bar test renders correctly 1`] = `
Array [
  <div
    className="booking-page__container__item__title"
  >
    <h2>
      Payemnt methods
    </h2>
  </div>,
  <div
    className="payment-booking"
  >
    <div
      className="ant-radio-group ant-radio-group-outline"
    >
      <label
        className="ant-radio-wrapper"
      >
        <span
          className="ant-radio"
          style={Object {}}
        >
          <input
            checked={false}
            className="ant-radio-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            type="radio"
            value={0}
          />
          <span
            className="ant-radio-inner"
          />
        </span>
        <span>
          Paypal
        </span>
      </label>
    </div>
    <div
      className="BookingPayment-content"
    />
  </div>,
]
`;
